### ISSUE TITLE
Tensorflow BERT LAMB optimizer algorithm problem

### ISSUE THREAD

**Issue by LiweiPeng** [NVIDIA/DeepLearningExamples#215]
In Tensorflow BERT LAMB optimizer code (https://github.com/NVIDIA/DeepLearningExamples/blob/master/TensorFlow/LanguageModeling/BERT/optimization.py#L295), the LAMB algorithm is from the LAMB paper (https://arxiv.org/abs/1904.00962) version 1, not from LAMB paper version 3.  I used LAMB version 3 and found it work great. What's the reason to prefer version 1 not version 3? Thanks.

**Comment by sharathts** [comment]
We implemented LAMB when version 1 was out and it didn't work out of the box. We had to make a couple of changes to the original version 1 algorithm, for example gradient normalization/clipping, warmup etc ., We have been using this implementation since and haven't tried later version.

Could you share data on how version 3 performed better than v1 for you if any? Thanks.

**Comment by LiweiPeng** [comment]
@sharathts Thanks for the quick reply. I didn't use version 1 for large batch size training. I only used the version 3. So not sure which one is better. I do see that version 3 is quite different from version 1 in terms of implementation.

**Comment by LiweiPeng** [comment]
@sharathts I'd like to use your lamb implementation. You mentioned version1 lamb didn't work out of box. Can you provide more details what parameters need to tune and how to tune generally to use version1 lamb? for example, how do you tune warmup, learning rate differently from standard bert?

**Comment by LiweiPeng** [comment]
I found these numbers in the source code scripts. I'll close this issue
